docs/aws/audit/redshiftmonitoring/rules/redshift_instance_generation.mdx

```diff
@@ -24,6 +24,224 @@ CBP
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the Redshift resources.
+
+2. Once the AWS CLI is set up, you can list all the Redshift clusters using the following command:
+
+   ```
+   aws redshift describe-clusters --query 'Clusters[*].[ClusterIdentifier,NodeType]'
+   ```
+   This command will return a list of all Redshift clusters along with their node types.
+
+3. To check if the Redshift clusters are using the latest generation of nodes, you need to compare the NodeType of each cluster with the latest generation node types. As of now, the latest generation node types are 'dc2' and 'ra3'.
+
+4. You can use a Python script to automate this process. The script will use the AWS SDK (boto3) to list all the Redshift clusters and their node types, and then compare them with the latest generation node types. Here is a simple example:
+
+   ```python
+   import boto3
+
+   # Create a Redshift client
+   redshift = boto3.client('redshift')
+
+   # List all Redshift clusters
+   clusters = redshift.describe_clusters()
+
+   # Check if the clusters are using the latest generation of nodes
+   for cluster in clusters['Clusters']:
+       if cluster['NodeType'] not in ['dc2', 'ra3']:
+           print(f"Cluster {cluster['ClusterIdentifier']} is not using the latest generation of nodes.")
+   ```
+   This script will print the identifiers of the clusters that are not using the latest generation of nodes.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. This will allow you to interact with AWS services including Redshift.
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your AWS credentials.
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2'
+)
+```
+
+3. Create a Redshift client using the session.
+
+```python
+redshift = session.client('redshift')
+```
+
+4. Use the `describe_clusters` method to retrieve information about all Redshift clusters. Check the `NodeType` attribute of each cluster to determine if it's using the latest generation of nodes. If the `NodeType` is not one of the latest generation node types (e.g., 'dc2.large', 'dc2.8xlarge', 'ra3.4xlarge', 'ra3.16xlarge'), then the cluster is misconfigured.
+
+```python
+response = redshift.describe_clusters()
+
+for cluster in response['Clusters']:
+    if cluster['NodeType'] not in ['dc2.large', 'dc2.8xlarge', 'ra3.4xlarge', 'ra3.16xlarge']:
+        print(f"Cluster {cluster['ClusterIdentifier']} is not using the latest generation of nodes.")
+```
+
+This script will print out the identifiers of all clusters that are not using the latest generation of nodes.
+</Accordion>
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>